calour/io.py

```diff
@@ -235,6 +239,13 @@ def _read_metadata(ids, f, kwargs):
             logger.warning('These have data but do not have metadata: %r' % diff)
         # reorder the id in metadata to align with biom
         # metadata = metadata.loc[ids, ]
+
+        # check if we have duplicate index ids (which will raise)
+        dup = metadata.index.duplicated(keep=False)
+        if dup.sum() > 0:
+            errmsg = '%d duplicate id values encountered in column %s of mapping file %r:' % (dup.sum(), metadata.index.name, f)
```

Comment:
metadata.index.name can be empty. it's better to say "1st column" or "index column"

Reply:
changed to take care of both cases :)